ArduCopter/ArduCopter.cpp

```diff
@@ -594,5 +597,18 @@ void Copter::publish_osd_info()
     osd.set_nav_info(nav_info);
 }
 #endif
-
+#if FRSKY_TELEM_ENABLED == ENABLED
+void Copter::update_nav_info()
+{
+    AP_Frsky_Telem::NavInfo nav_info;
+    //
+    nav_info.wp_distance = flightmode->wp_distance() * 1.0e-2f;
```

Comment:
How are you determining the applicability of this data on the other end?

wp_distance means a lot of things in a lot of modes.... just not all the same.

Reply:
Yes, wp_distance means a lot of different things, I noticed it in my tests in RTL for instance and I'm sure each mode has his own use of the waypoint "subsystem", this though is not as bad as it might seem from "the other end" point of view since the library already sends frametype and flight modes. In my tests I added some lua code to properly display the data in a proper context.
The basic idea is to send data as raw as possible and eventually process it (apply some logic) at the rendering side.

Reply:
I might be missing a mode or two here but, upon a cursory reading, I find that wp_distance(), as it relates to its use in Copter only, gets us either:
- horizontal distance to destination in cm for RTL, SmartRTL, Auto, and Guided_WP modes
- horizontal distance to position target in cm for Loiter, Circle, and Guided_PosVel modes
That doesn't seem to represent a lot of different things. What am I missing?
By the way, Alessandro is right in that flightmode->wp_distance() is used by Copter::send_nav_controller_output() which is what is being sent to GCS via MAVLink.